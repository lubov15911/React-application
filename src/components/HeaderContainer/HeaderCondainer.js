import React from 'react';
import PropTypes from 'prop-types';

import HomePageLogoButton from '../HomePageLogoButton/HomePageLogoButton';
import HomePageSearchButton from '../HomePageSearchButton/HomePageSearchButton';
import SearchHeader from '../SearchHeader/SearchHeader';
import MovieCard from '../MovieCard/MovieCard';
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';
import './HeaderContainer.scss';

const HeaderContainer = (props) => {
    const {
        routeToHomePage,
        movieId,
        error,
    } = props;
    return (
        <div className="header-container">
            <div className="buttons-container">
                <HomePageLogoButton handleClick={routeToHomePage} />
                {movieId && <HomePageSearchButton handleClick={routeToHomePage} />}
            </div>
            {!movieId ? <ErrorBoundary><SearchHeader error={error}/></ErrorBoundary> : <MovieCard />}
        </div>
    );
};
HeaderContainer.propTypes = {
    movieId: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
    // eslint-disable-next-line react/require-default-props
    error: PropTypes.bool,
    routeToHomePage: PropTypes.func.isRequired,
};

export default HeaderContainer;
