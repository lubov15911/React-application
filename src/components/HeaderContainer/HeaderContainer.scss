.header-container {
  background: #232323;
  display: flex;
  flex-direction: column;
  padding: 3em 2em 4em 6em;

  .buttons-container {
    display: flex;
    justify-content: space-between;

    .home-page-btn {
      color: #f65261;
      font-size: 14px;
    }
  }
}
