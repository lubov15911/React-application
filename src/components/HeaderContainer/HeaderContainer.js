// @flow

import React from 'react';
import { Link, useLocation } from 'react-router-dom';

import SearchContainer from '../SearchContainer';
import MovieCard from '../MovieCard';

import './HeaderContainer.scss';

const HeaderContainer = () => {
    const location = useLocation();
    let renderHeader = null;
    let isMovie = false;

    if (location.pathname.startsWith('/film')) {
        renderHeader = <MovieCard />;
        isMovie = true;
    } else {
        renderHeader = <SearchContainer />;
    }

    return (
        <div className="header-container">
            <div className="buttons-container">
                <Link to="/" className="home-page-btn"><b>netflix</b>roulette</Link>
                {isMovie && <Link to="/" className="home-page-btn">Search icon</Link>}
            </div>
            {renderHeader}
        </div>
    );
};

export default HeaderContainer;
