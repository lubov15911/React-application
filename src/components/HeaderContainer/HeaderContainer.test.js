import React from 'react';
import { shallow } from 'enzyme';

import HeaderContainer from './HeaderContainer';

describe('HeaderContainer', () => {
    it('should render correctly', () => {
        const fakeFunction = () => {};

        const component = shallow(<HeaderContainer />);

        expect(component).toMatchSnapshot();
    });
});
