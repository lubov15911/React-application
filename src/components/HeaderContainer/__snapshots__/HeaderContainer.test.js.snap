// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer should render correctly 1`] = `
<div
  className="header-container"
>
  <div
    className="buttons-container"
  >
    <a
      className="home-page-btn"
      href="."
    >
      <b>
        netflix
      </b>
      roulette
    </a>
  </div>
  <SearchContainer
    handleSearchValue={null}
    searchOption={null}
    toggleSearch={null}
  />
</div>
`;
